### PR TITLE
2023's viro rework: removing viro

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
This pull request is focused on removing virology as a job in a non-destructive manner, and changing virology mechanics to reflect its removal 

CONFIG CHANGES
the following changes can be toggled on or off in config, for friendliness to downstreams
- option to enable/disable virology as a job and map location. This uses a special locker for items, disables most items in viro fridge, and adds a mapping helper that turns certain pieces of non-viro scenery to viro objects, and sets doors to viro names/access
-For mappers,  virology needs: a virology mapping helper over each access airlock(maints included), which should be labelled PHARMACY or somesuch, a mapping helper over a smoke machine, a mapping helper over a potted plant, and a  l3closet/virology/starter. There still needs to be a preloaded viro smartfridge, a monkey enclosure, and a pandemic machine, but the area should be freely accessible to all medical personell. 
--option to change whether necrotic metabolism causes diseases to process while in corpses

the following changes exist to make the customizability of viruses heavily tunable. currently, they're *somewhat* customizeable (currently, with enabled options, you can combine viruses and remove some symptoms, and neuter others. heavily rng dependent based on what viruses spawn and what symptoms can be removed (symptom removal uses rng fixed at roundstart), making spreadsheet viro nearly impossible most rounds.)
- option to allow chemical customization of viruses to be disabled (turning this off also replaces the virus sample crate with a crate full of a few basic isolated symptoms). This is now enabled
-option to limit how many symptoms can be removed from a virus. At this time, I have synaptizine no longer function on viruses with four or less symptoms
-option to make symptom removal "Seeded"- every round, symptom IDS are scrambled, and the lowest ID symptoms are always removed from a virus first. this makes rng-based isolation with tenacity impossible. this is now enabled
-  option to disable extrapolator isolation. this is now enabled.
-  option to disable mixing viruses together. not enabled.
- option to  allow neutering and other utility virus recipes to be disabled. not currently enabled

the following changes exist to make random viruses more impactful, and make virology a bit easier to compensate for a lack of fine-tunability.
- option to change whether or not rats and dormant diseases infect players. these features are annoying and make viruses a bigger danger, making eating rats or sharing syringes dangerous. These features are enabled.
- option to allow certain "secret effects" on symptoms, such as hemetophagy vampires or autophageocytosis zombies. enabled
-  option to make spread symptoms always spread a virus. disabled.
-  option to make some virus thresholds easier to reach. tentatively enabled.
- option to make the above three options only apply to random event viruses (also making said viruses uncustomizable). enabled.

Non- config changes

- disease symptoms order themselves based on the roundstart removal seed, to remove guesswork and frustration. First symptom is removed first.
-  improve random diseases, so they always have a transmission symptom and a symptom with a notable effect
-change how extrapolators are obtained. now, they're obtained through science
- [ ] revamp virus cure list
- remove dermagraphic ovulogenesis (pointless symptom i hate it)
- re-adds viro zombies. These are now nerfed, doing less damage, and spreading their virus rather than spreading zombie tumors. 

## Why It's Good For The Game
virology as it is is extremely bad for roleplay. The virology department is poorly equipped to deal with actual outbreaks, and spreading beneficial diseases adds little positive to the game. This means the department is either powergame fuel for people who spend hours making out of game spreadsheets, or grief fuel and banbait. often both.

this style of removal still leaves enough disease customization in to engage players who wish to engage with it, but does not reward a hefty array of spreadsheets and recipes, instead encouraging a more intuitive grasp of the system and how symptoms mesh together

Whether or not I make this PR, viro will be removed, so I'd prefer to be the one to kill it. with this approach, downstreams will still be able to enjoy beeviro, without worrying about upstream nerfs or removals disturbing it

## Testing Photographs and Procedure
-draft-
<details>
I did not  remember to screenshot my testing, but it consisted of testing a few event viruses, the mapping helpers, mixing viruses till i got the seeding to work right, testing dormant diseases, etc.

</details>

## Changelog
:cl:
del: removes the virologist role. this is config based.
del: Removes most means to customize viruses This is config based. Currently, you can remove up to 2 symptoms from a disease, neuter symptoms, and mix viruses. Symptom removal is not random, with each symptom being removed based on a value given at roundstart.
del: dermagraphic ovulogenesis symptom
tweak: while virus customization is disabled, virus sample crates instead ship 5 isolated utility symptoms
add: re-adds viro zombies. They're now weaker, and only spread their diseases, instead of spreading the tumor itself.
add: adds a new chemical reaction, which allows you to remove faltered, immutable, dormant, and carrier status from viruses
balance: random viruses made through events should be more impactful now
balance: some symptom thresholds should be far easier to reach. config controlled. 
mapping: adds tools for mappers to make virology departments a pharmaceutical research lab accessible by all medical personelll unless viro is enabled
tweak: toxoplasmosis sapiens is now level 0, and can *technically* appear, but, like all level 0 symptoms, will practically never appear, and will be even less likely to have any significant effect. However, the owo speak effect has been substantially toned down, and should now only impede the speech of certain letters and induce random cat noises, rather than replacing words arbitrarily.
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
not, of course, ready for review

**Comment by zeskorion** [comment]
I've finished everything but the cure revamp, which is not as important. I have not tested anything yet, however. Should also port @Absolucy 's changes to necrotic adaptation to this.

**Comment by zeskorion** [comment]
just added a buncha .stat checks for necrotic metabolism. shouldn't proc to_chats now. A few symptoms totally have it disabled, especially problematic instances.

**Comment by zeskorion** [comment]
this pr is now as close to ready as it's gonna get without a TM to work out any kinks that may exist, please tell me if/when i need to make wiki changes

**Comment by zeskorion** [comment]
tfw random failure due to unrelated code

**Comment by EvilDragonfiend** [comment]
yeah it sometimes happens. really can't fix it for now. pushing anything fixes it or maints can rerun the test

**Comment by Job-IDev** [comment]
I'm just gonna be straight about this. If we truly hate viro so much why not rm -rf it?

I've attempted to play some virology on low pop yesterday and it was horrible.
Without a limitless cargo/medbay budget you have basically become worthless to the station.

The chance of finding a good pair or an isolated symptom with your extrapolator is so slim it ain't even worth looking for them. Whilst you can order them, you have to hope cargo is also willing to spend money on your shipment. (Which previously wasn't the case in the majority of cases)

What might be the worst part about this update is that, instead of solving the issue that virologists are not interacting with the station for RP, this update basically binds the viro to their lab for the rest of the shift with exactly no valuable output. (At least if they are determined enough) 

An overall problem with the role in combination with RP, is that medbay has too many ways to resolve a virus issue without ever talking to the virologist.
I've personally had too many rounds where there was finally some kind of virus outbreak (without a viro spreading it) only to hear about it after the round ended. Simply because the combination of chemistry and a health analyzer could solve the matter.

To summarize, this PR will, in my eyes, not fix the core issues of virology. Despite the good effort of trying to fix this lingering problem, I personally think that it might be better to rip the bandaid of and integrating virus curing into the medical doctors' role.

Thank you for listening to my Ted Talk. It's time for me to find a different role to ~~get removed~~ play.

**Comment by zeskorion** [comment]
> I'm just gonna be straight about this. If we truly hate viro so much why not rm -rf it?
> 
> I've attempted to play some virology on low pop yesterday and it was horrible. Without a limitless cargo/medbay budget you have basically become worthless to the station.
> 
> The chance of finding a good pair or an isolated symptom with your extrapolator is so slim it ain't even worth looking for them. Whilst you can order them, you have to hope cargo is also willing to spend money on your shipment. (Which previously wasn't the case in the majority of cases)
> 
> What might be the worst part about this update is that, instead of solving the issue that virologists are not interacting with the station for RP, this update basically binds the viro to their lab for the rest of the shift with exactly no valuable output. (At least if they are determined enough)
> 
> An overall problem with the role in combination with RP, is that medbay has too many ways to resolve a virus issue without ever talking to the virologist. I've personally had too many rounds where there was finally some kind of virus outbreak (without a viro spreading it) only to hear about it after the round ended. Simply because the combination of chemistry and a health analyzer could solve the matter.
> 
> To summarize, this PR will, in my eyes, not fix the core issues of virology. Despite the good effort of trying to fix this lingering problem, I personally think that it might be better to rip the bandaid of and integrating virus curing into the medical doctors' role.
> 
> Thank you for listening to my Ted Talk. It's time for me to find a different role to ~get removed~ play.

if this is tmed and configured correctly, you should not be playing virologist at all, as it disables the role entirely. the virus system exists independently of the job

**Comment by JixS4v** [comment]
> I'm just gonna be straight about this. If we truly hate viro so much why not rm -rf it?
> 
> I've attempted to play some virology on low pop yesterday and it was horrible. Without a limitless cargo/medbay budget you have basically become worthless to the station.
> 
> The chance of finding a good pair or an isolated symptom with your extrapolator is so slim it ain't even worth looking for them. Whilst you can order them, you have to hope cargo is also willing to spend money on your shipment. (Which previously wasn't the case in the majority of cases)
> 
> What might be the worst part about this update is that, instead of solving the issue that virologists are not interacting with the station for RP, this update basically binds the viro to their lab for the rest of the shift with exactly no valuable output. (At least if they are determined enough)
> 
> An overall problem with the role in combination with RP, is that medbay has too many ways to resolve a virus issue without ever talking to the virologist. I've personally had too many rounds where there was finally some kind of virus outbreak (without a viro spreading it) only to hear about it after the round ended. Simply because the combination of chemistry and a health analyzer could solve the matter.
> 
> To summarize, this PR will, in my eyes, not fix the core issues of virology. Despite the good effort of trying to fix this lingering problem, I personally think that it might be better to rip the bandaid of and integrating virus curing into the medical doctors' role.
> 
> Thank you for listening to my Ted Talk. It's time for me to find a different role to ~get removed~ play.

Virologist as a job should have been completely axed to my knowledge, you shouldn't be able to join as the virologist in theory

**Comment by Job-IDev** [comment]
> Virologist as a job should have been completely axed to my knowledge, you shouldn't be able to join as the virologist in theory

As for ~2 hours ago I was still able to select the virologist role. So there might be something wrong with the configuration.

**Comment by zeskorion** [comment]
> > Virologist as a job should have been completely axed to my knowledge, you shouldn't be able to join as the virologist in theory
> 
> As for ~2 hours ago I was still able to select the virologist role. So there might be something wrong with the configuration.

ah, i forgot to comment allow_virologist after testing

@EvilDragonfiend can you make sure allow_virologist is commented as well

**Comment by Fronsis** [comment]
Zombie midround event is still possible according to the code right?

**Comment by zeskorion** [comment]
> Zombie midround event is still possible according to the code right?

correct

**Review by EvilDragonfiend — APPROVED** [review]
Testmerge passed. I see nothing particularly wrong.
The issue here is that the code has a lot of config options to make it hardly readible, but I guess it's a fair cost since we decided to keep it configurable with our complex feeling we have - we hate them and we love them.

**Comment by EvilDragonfiend** [comment]
Do not merge until Crossed approves the configuration.

**Comment by PowerfulBacon** [comment]
These are only basic config additions and has no chance to change anything on the hosting/server side, crossed doesn't really need to ensure their correctness
